### ISSUE TITLE
Sync progress counters do not work with migrated Realm files

### ISSUE THREAD

**Issue by switchawk** [realm/realm-mobile-platform#119]
I'm using the new sync notifications for the purpose of a sync indicator when any upload or download is taking place.

I'm doing this in the following way:

```
        let session = SyncUser.current?.session(for: Constants.syncServerURL)!
        self.uploadToken = session!.addProgressNotification(for: .upload,
                                             mode: .reportIndefinitely)
        { [weak self] progress in
            print("progress = \(progress)")
            if progress.isTransferComplete {
                self?.isUploading = false
            } else {
                self?.isUploading = true
            }
            ...
        }
```
With no other clients running I expected that little or nothing will be synced every launch. I get this output every launch:

```
progress = Progress(transferredBytes: 0, transferrableBytes: 184320)
```

And it never gets called again. Tried replacing `.reportIndefinitely` to `.forCurrentlyOutstandingWork` and got the same results.

Side note: I had a download token as well before. Mentioning this as it may be related to this issue on the server side. Anyway, that code was removed and the issue persists.

Realm Object Server Version: 1.0.0
Flavor: Developer
Server OS & Version: Ubuntu 16.04.1 x64
Client SDK Version: 2.4.2 (iOS Simulator)
Client OS & Version: 10.12.3

**Comment by switchawk** [comment]
Update: It seems the connection doesn't get through to the server, according to the logs. There's zero activity on the server. How can I debug this further on the client?

**Comment by bigfish24** [comment]
Sorry about this! I have assigned two engineers on our team to look into the issue.

**Comment by austinzheng** [comment]
Hello! Sorry to hear you're having problems.

To further troubleshoot, could you try two things?
* Can you add an error handler to your sync manager? Something like the following, which prints out any errors the client-side library encounters: `SyncManager.shared.errorHandler = { error, _ in print("Received an error: \(error)") }`
* Can you set the logging level on the sync manager to `.all` or `.trace`? `SyncManager.shared.logLevel = SyncLogLevel.all`

Hopefully either or both of these steps will give us more information to work from.

**Comment by switchawk** [comment]
Here's the output I got: (I put ... in some places)
```
2017-02-02 20:40:18.567 MyApp[49836:2170967] [Crashlytics] Version 3.8.3 (120)
2017-02-02 20:40:18.762 MyApp[49836:2171193] Sync: Opening Realm file: ...
progress = Progress(transferredBytes: 0, transferrableBytes: 694)
2017-02-02 20:40:19.377 MyApp[49836:2171193] Sync: Using already open Realm file: ...
2017-02-02 20:40:19.379 MyApp[49836:2171193] Sync: Connection[1]: WebSocket::Websocket()
2017-02-02 20:40:19.380 MyApp[49836:2171193] Sync: Connection[1]: Reconnecting in 0 milliseconds
2017-02-02 20:40:19.380 MyApp[49836:2171193] Sync: Using already open Realm file: ...
2017-02-02 20:40:19.381 MyApp[49836:2171193] Sync: Connection[1]: Session[1]: Starting session for '...'
2017-02-02 20:40:19.381 MyApp[49836:2171193] Sync: Connection[1]: Session[1]: last_version_available  = 13
2017-02-02 20:40:19.381 MyApp[49836:2171193] Sync: Connection[1]: Session[1]: progress_server_version = 0
2017-02-02 20:40:19.381 MyApp[49836:2171193] Sync: Connection[1]: Session[1]: progress_client_version = 0
2017-02-02 20:40:19.382 MyApp[49836:2171193] Sync: Connection[1]: Resolving '...:9080'
progress = Progress(transferredBytes: 0, transferrableBytes: 694)
2017-02-02 20:40:19.383 MyApp[49836:2171193] Sync: Connection[1]: Connecting to endpoint '...:9080' (1/1)
2017-02-02 20:40:19.617 MyApp[49836:2171193] Sync: Connection[1]: Connected to endpoint '...:9080' (from '...:55315')
2017-02-02 20:40:19.617 MyApp[49836:2171193] Sync: Connection[1]: WebSocket::initiate_client_handshake()
2017-02-02 20:40:19.619 MyApp[49836:2171193] Sync: Connection[1]: HTTP request =
GET /realm-object-server HTTP/1.1
Connection: Upgrade
Host: 138.197.211.135
Sec-WebSocket-Key: oRhQFMVJoNO1bh9Q1+cQSg==
Sec-WebSocket-Protocol: io.realm.protocol
Sec-WebSocket-Version: 13
Upgrade: websocket
X-Realm-Access-Token: ...
X-Realm-Path: ....
X-Realm-Protocol-Version: 16

2017-02-02 20:40:19.874 MyApp[49836:2171193] Sync: Connection[1]: WebSocket::handle_http_response_received()
2017-02-02 20:40:19.875 MyApp[49836:2171193] Sync: Connection[1]: HTTP: Failed to switch protocols (response status = 400)
2017-02-02 20:40:19.875 MyApp[49836:2171193] Sync: Connection[1]: Connection closed due to error
2017-02-02 20:40:19.876 MyApp[49836:2171193] Sync: Connection[1]: Reconnecting in 3076739 milliseconds
```

**Comment by morten-krogh** [comment]
I believe, the problem is that you have migrated the Realm from before Sync notifications were
introduced. The "old" Realm can work with the progress notifications, but the "transferredBytes"
from before the introduction of sync progress are not counted correctly.

transferredBytes works by increasing a counter every time a Download message is 
received. Unfortunately, the counter does not know what happened with that particular Realm
file before we introduced the progress notifications.  

With a new device or by reinstalling the app, it will exactly work, as you see.

You can check if this is correct by checking that the data actually is on the device.

I know this is silly, and I apologize.

**Comment by radu-tutueanu** [comment]
This should be fixed for Realm Cocoa > `2.4.3`
